api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/TextMessageProcessor.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.dhsc.htbhf.claimant.message.processor;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+import uk.gov.dhsc.htbhf.claimant.entity.Message;
+import uk.gov.dhsc.htbhf.claimant.exception.EventFailedException;
+import uk.gov.dhsc.htbhf.claimant.message.MessageStatus;
+import uk.gov.dhsc.htbhf.claimant.message.MessageType;
+import uk.gov.dhsc.htbhf.claimant.message.MessageTypeProcessor;
+import uk.gov.dhsc.htbhf.claimant.message.context.MessageContextLoader;
+import uk.gov.dhsc.htbhf.claimant.message.context.TextMessageContext;
+import uk.gov.dhsc.htbhf.claimant.service.audit.FailedTextEvent;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+import uk.gov.service.notify.SendSmsResponse;
+
+import java.util.UUID;
+
+@Slf4j
+@Component
+public class TextMessageProcessor implements MessageTypeProcessor {
+
+    private final NotificationClient client;
+    private final MessageContextLoader messageContextLoader;
+
+    public TextMessageProcessor(NotificationClient client,
+                                MessageContextLoader messageContextLoader) {
+        this.client = client;
+        this.messageContextLoader = messageContextLoader;
+    }
+
+    @Override
+    public MessageStatus processMessage(Message message) {
+
+        TextMessageContext textMessageContext = messageContextLoader.loadTextMessageContext(message);
+        UUID textReference = UUID.randomUUID();
```

Comment:
I'm just thinking should we start using claimId as the reference instead, this is if in future if we need to make a call to check the status of the message we are able to do so using something that we hold in the database. https://docs.notifications.service.gov.uk/java.html#get-the-status-of-multiple-messages